tests/unit/runtime/test_multiple_models.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import pytest
+import deepspeed
+import deepspeed.comm as dist
+import torch
+from unit.common import DistributedTest
+from unit.simple_model import SimpleModel, random_dataloader
+
+
+def create_model(config_dict):
+    hidden_dim = 64
+    model = SimpleModel(hidden_dim)
+    model, _, _, _ = deepspeed.initialize(model=model, model_parameters=model.parameters(), config=config_dict)
+    return model
+
+
+def train_loop(num_models, config_dict, dtype):
+    hidden_dim = 64
+
+    models = [create_model(config_dict) for _ in range(num_models)]
+    data_loader = random_dataloader(model=models[0],
+                                    total_samples=4,
+                                    hidden_dim=hidden_dim,
+                                    device=models[0].device,
+                                    dtype=dtype)
+    dist.barrier()
+    for _, batch in enumerate(data_loader):
+        losses = [m.module(batch[0], batch[1]) for m in models]
+        loss = sum(l / (i + 1) for i, l in enumerate(losses))
+        loss.backward()
+
+        for m in models:
+            m._backward_epilogue()
```

Comment:
so will this be now a requirement for users wanting to use 2+ models? If so it needs to be documented, no?

Reply:
Yes, it needs documenting as experimental since I view it as a temporary hack. The actual solution is a hook that triggers epilogue after the last grad creation of the model. Althought Z3 has similar code, I have not found time to adapt it :(

Reply:
yeah, let's document that - otherwise what's the point of investing so much energy if others then don't know they can use it. Does it resonate?

Even a super basic example or even pointing to a test would be better than nothing, IMHO.

Reply:
Thank you for adding the docs, Tunji!